docs/developer/index.rst

```diff
@@ -98,6 +98,44 @@ run allows us to generate a coverage report after that::
 The report is in ``report/index.html``, which links to individual
 files and shows exactly which lines were excuted while running the tests.
 
+Run doctests locally
+--------------------
+If `doctestplus <https://pypi.org/project/pytest-doctestplus/>` is installed
+(and it probably is because it's part of
+`sphinx-astropy <https://pypi.org/project/sphinx-astropy/>`,
+which is required to build the documentation locally), you can
+run the examples in the documentation or in the doctrings of individual
+functions, for example:
```

Comment:
```suggestion
functions, for example::
```

There's a problem with the current documentation since the following `pytest` lines do not get written as separate lines. I assume that adding a colon here will fix it, but this is just a guess...

![screenshot](https://user-images.githubusercontent.com/224638/199530549-226419cd-288e-4090-9162-9c3631ab3e2c.png)

Reply:
So, I ran

    pytest --doctest-plus

and it collected 7623 tests, which is the same number as I got from just `pytest`. Is this expected?

As a further check,

    pytest --doctest-plus sherpa/astro/data.py
    pytest sherpa/astro/data.py

both run 13 tests. So how do I know that it's actually running the extra tests - do I just have to assume it is?

Reply:
To answer the first part of this: With the changes in this PR, doctest-plus is automatically activated for all files that are NOT on the doctestplus blacklist. You can see that the number of tests increases compared to main.
Note that the number shown in my output below is different from what @DougBurke said above, but that's OK because (a) the number of tests run depends on the plotting/IO backend and (b) the number of tests has changed in main when I just rebased locally (not pushed yet). The point is that the settings in this PR increas the number of tests compared to main. The increase is 6, because there are 6 files in the rst files in the docs that are **NOT** blacklisted in `pytest.ini` and doctest run each rst file as a single test.
On main
```
(sherpaalone) MoritzAirRoseGold ~/code/sherpa> pytest
================================================================================================== test session starts ==================================================================================================
platform darwin -- Python 3.10.0, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /Users/guenther/code/sherpa, configfile: pytest.ini
plugins: doctestplus-0.11.2, typeguard-2.13.3, xvfb-2.0.0
collecting ... WARNING: Failed to import any usable sherpa.plotting backend. Plotting routines will not be available.
collected 7547 items       
```

On this branch:
```
(sherpaalone) MoritzAirRoseGold ~/code/sherpa> pytest 
================================================================================================== test session starts ==================================================================================================
platform darwin -- Python 3.10.0, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /Users/guenther/code/sherpa, configfile: pytest.ini
plugins: doctestplus-0.11.2, typeguard-2.13.3, xvfb-2.0.0
collecting ... WARNING: Failed to import any usable sherpa.plotting backend. Plotting routines will not be available.
collected 7555 items                                                                                         ```